tensorboard/plugins/debugger_v2/debug_data_multiplexer.py

```diff
@@ -136,56 +115,73 @@ def __init__(self, logdir):
-        separate thread immediately.
+        If the reader has already been created, a new one will not be created and
+        this function is a no-op.
 
         If a reader has not been created, create it and start periodic calls to
         `update()` on a separate thread.
         """
+        if self._reader:
+            return
         with self._reader_lock:
             if not self._reader:
                 try:
+                    # TODO(cais): Avoid conditional imports and instead use
+                    # plugin loader to gate the loading of this entire plugin.
                     from tensorflow.python.debug.lib import debug_events_reader
                     from tensorflow.python.debug.lib import (
                         debug_events_monitors,
                     )
-
-                    self._reader = debug_events_reader.DebugDataReader(
-                        self._logdir
-                    )
-                    self._monitors = [
-                        debug_events_monitors.InfNanMonitor(
-                            self._reader, limit=DEFAULT_PER_TYPE_ALERT_LIMIT
-                        )
-                    ]
-                    # NOTE(cais): Currently each logdir is enforced to have only one
-                    # DebugEvent file set. So we add hard-coded default run name.
-                    run_repeatedly_in_background(self._reader.update)
-                    # TODO(cais): Start off a reading thread here, instead of being
-                    # called only once here.
                 except ImportError:
                     # This ensures graceful behavior when tensorflow install is
                     # unavailable or when the installed tensorflow version does not
                     # contain the required modules.
-                    pass
+                    return
+
+                try:
+                    self._reader = debug_events_reader.DebugDataReader(
+                        self._logdir
+                    )
                 except AttributeError:
                     # Gracefully fail for users without the required API changes to
                     # debug_events_reader.DebugDataReader introduced in
                     # TF 2.1.0.dev20200103. This should be safe to remove when
                     # TF 2.2 is released.
-                    pass
+                    return
                 except ValueError:
                     # When no DebugEvent file set is found in the logdir, a
                     # `ValueError` is thrown.
-                    pass
-            else:
-                _timer.wake()
+                    return
+
+                self._monitors = [
+                    debug_events_monitors.InfNanMonitor(
+                        self._reader, limit=DEFAULT_PER_TYPE_ALERT_LIMIT
+                    )
+                ]
+                # NOTE(cais): Currently each logdir is enforced to have only one
```

Comment:
I'm not sure I understand this comment - where is the hard-coded default run name?

Reply:
This comment is out of place. It's also redundant with the doc string of `FirstEventTimestamp` and `DEFAULT_DEBUGGER_RUN_NAME` in this same file. Removed it.